wallet-new/src/Cardano/Wallet/Kernel/BListener.hs

```diff
@@ -49,21 +61,117 @@ prefilterBlock' pw b = do
       let (conMap, conMeta) = mconcat $ map (uncurry (prefilterBlock b)) ws
       in ((b ^. rbContext, conMap), conMeta)
 
+data BackfillFailed
```

Comment:
We should probably document that the only error we can except under normal circumstances is `CouldNotReachCheckpoint` (if we missed a rollback); `CoindNotFindBlockForHeader` and `NotAMainBlock` indicate database corruption; `SuccessorChanged` under normal circumstances should never happen (the wallet worker sequentializes apply block and switch to fork), but _might_ happen very rarely (like when there is a reset database state for instance?).